### PR TITLE
Fix connected components issue2258

### PR DESCRIPTION
**Pull request by wtwhite**
The `_values` member was being resized to the number of vertices currently in the graph, which works if vertices have never been deleted.  But if vertices have ever been deleted, then there might exist vertices in the graph with higher IDs, and trying to look these up in `_values` later on would cause an assertion failure or segfault.

After this fix, the MWE in issue #2258 correctly reports 1 component in the graph instead of crashing.

### CODE REVIEW

**Comment by h-2** [comment]
@smehringer Can you have a look?

**Comment by h-2** [comment]
@wtwhite Thanks for the PR!

**Comment by h-2** [comment]
Reassigning to @joergi-w as he will take over maintainership of graphs in SeqAn!

File: include/seqan/misc/union_find.h [hunk 1]
```diff
@@ -298,7 +298,8 @@ resizeVertexMap(UnionFind<TValue> & unionFind,
                 Graph<TSpec> const & g)
 {
     clear(unionFind);
-    return resize(unionFind, numVertices(g));
+    resizeVertexMap(unionFind._values, g, -1);
+    return length(unionFind);		// Note: Other resizeVertexMap() overloads return void...
```
Comment (joergi-w) [inline include/seqan/misc/union_find.h]: True. I might change this function to `void` in the future for consistency and the returned value is not used throughout the SeqAn library anyway. Thank you for pointing that out.

Comment (wtwhite) [reply include/seqan/misc/union_find.h]: You're welcome :)

**Pull request merged**